src/Ng/Lightning.cs

```diff
@@ -0,0 +1,397 @@
+
+                    var latestCatalogPages = new Dictionary<string, Uri>();
+
+                    foreach (CatalogIndexEntry catalogIndexEntry in packageRegistrationGroup)
+                    {
+                        string key = catalogIndexEntry.Version.ToNormalizedString();
+                        if (latestCatalogPages.ContainsKey(key))
+                        {
+                            latestCatalogPages[key] = catalogIndexEntry.Uri;
+                        }
+                        else
+                        {
+                            latestCatalogPages.Add(key, catalogIndexEntry.Uri);
+                        }
+
+                        if (latestCommit < catalogIndexEntry.CommitTimeStamp)
+                        {
+                            latestCommit = catalogIndexEntry.CommitTimeStamp;
+                        }
+                    }
+
+                    foreach (var latestCatalogPage in latestCatalogPages)
+                    {
+                        streamWriter.WriteLine("{0}", latestCatalogPage.Value);
+                    }
+                }
+            }
+
+            log.WriteLine("Finished preparing lightning reindex file. Output file: {0}", indexFile);
+
+            // Write cursor to storage
+            log.WriteLine("Start writing new cursor...");
+
+            var account = CloudStorageAccount.Parse(storageAccount);
+            var storageFactory = (StorageFactory)new AzureStorageFactory(account, container);
+            var storage = storageFactory.Create();
+            var cursor = new DurableCursor(storage.ResolveUri("cursor.json"), storage, latestCommit);
+            cursor.Value = latestCommit;
+            await cursor.Save(CancellationToken.None);
+
+            log.WriteLine("Finished writing new cursor.");
+            
+            // Write command files
+            log.WriteLine("Start preparing lightning reindex command files...");
+
+            string templateFileContents;
+            using (var templateStreamReader = new StreamReader(templateFile))
+            {
+                templateFileContents = await templateStreamReader.ReadToEndAsync();
+            }
+
+            int batchNumber = 0;
+            int batchSizeValue = int.Parse(batchSize);
+            for (int batchStart = 0; batchStart < numberOfEntries; batchStart += batchSizeValue)
+            {
+                var batchEnd = (batchStart + batchSizeValue - 1);
+                if (batchEnd >= numberOfEntries)
+                {
+                    batchEnd = numberOfEntries - 1;
+                }
+
+                var cursorCommandFileName = "cursor" + batchNumber + ".cmd";
+                var cursorTextFileName = "cursor" + batchNumber + ".txt";
+
+                using (var cursorCommandStreamWriter = new StreamWriter(Path.Combine(outputFolder, cursorCommandFileName)))
+                using (var cursorTextStreamWriter = new StreamWriter(Path.Combine(outputFolder, cursorTextFileName)))
```

Comment:
how about we always include { } even after "using" I realize this moves the code to the right but still clearer in my mind.